Remove BUILD_NAMEDTENSOR from codegen and .cu files

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31117 Remove unnecessary ATen/core/EnableNamedTensor.h
* #31116 Remove all remaining usages of BUILD_NAMEDTENSOR
* **#31047 Remove BUILD_NAMEDTENSOR from codegen and .cu files**

Changelist:
- remove BUILD_NAMEDTENSOR from .cu files
- remove BUILD_NAMEDTENSOR special handling in function_wrapper.py,
gen.py
- remove BUILD_NAMEDTENSOR from cpp_extension.py. This code actually
did nothing because we always compile with BUILD_NAMEDTENSOR.

Future work:
- Remove torch._C.BUILD_NAMEDTENSOR. This is used in some python
functions.
- Remove aten/src/ATen/env.py, which defines BUILD_NAMEDTENSOR=True for
codegen. There should be no use cases of this left after this PR, but it
is future work to check that.
- Remove `aten/src/ATen/core/EnableNamedTensor.h`, which defines
the BUILD_NAMEDTENSOR C++ macro. There should be no uses of this left
after this PR, but it is future work to check that

Test Plan:
- run tests

Differential Revision: [D18908442](https://our.internmc.facebook.com/intern/diff/D18908442)

zou3519 committed:
Remove BUILD_NAMEDTENSOR from codegen and .cu files

Changelist:
- remove BUILD_NAMEDTENSOR from .cu files
- remove BUILD_NAMEDTENSOR special handling in function_wrapper.py
- remove BUILD_NAMEDTENSOR from cpp_extension.py. This code actually
did nothing because we always compile with BUILD_NAMEDTENSOR.

Test Plan:
- run tests

[ghstack-poisoned]

zou3519 committed:
Update on "Remove BUILD_NAMEDTENSOR from codegen and .cu files"


Changelist:
- remove BUILD_NAMEDTENSOR from .cu files
- remove BUILD_NAMEDTENSOR special handling in function_wrapper.py,
gen.py
- remove BUILD_NAMEDTENSOR from cpp_extension.py. This code actually
did nothing because we always compile with BUILD_NAMEDTENSOR.

Test Plan:
- run tests

Differential Revision: [D18908442](https://our.internmc.facebook.com/intern/diff/D18908442)

[ghstack-poisoned]

zou3519 committed:
Update on "Remove BUILD_NAMEDTENSOR from codegen and .cu files"


Changelist:
- remove BUILD_NAMEDTENSOR from .cu files
- remove BUILD_NAMEDTENSOR special handling in function_wrapper.py,
gen.py
- remove BUILD_NAMEDTENSOR from cpp_extension.py. This code actually
did nothing because we always compile with BUILD_NAMEDTENSOR.

Test Plan:
- run tests

Differential Revision: [D18908442](https://our.internmc.facebook.com/intern/diff/D18908442)

[ghstack-poisoned]

nairbv commented:
Can you also add what work remains for removing the flag after this PR (or if this is the final PR)?

zou3519 commented:
> Can you also add what work remains for removing the flag after this PR (or if this is the final PR)?

I'll add it to the body of the PR. There is some work remaining.

zou3519 committed:
Update on "Remove BUILD_NAMEDTENSOR from codegen and .cu files"


Changelist:
- remove BUILD_NAMEDTENSOR from .cu files
- remove BUILD_NAMEDTENSOR special handling in function_wrapper.py,
gen.py
- remove BUILD_NAMEDTENSOR from cpp_extension.py. This code actually
did nothing because we always compile with BUILD_NAMEDTENSOR.

Future work:
- Remove torch._C.BUILD_NAMEDTENSOR. This is used in some python
functions.
- Remove aten/src/ATen/env.py, which defines BUILD_NAMEDTENSOR=True for
codegen. There should be no use cases of this left after this PR, but it
is future work to check that.
- Remove `aten/src/ATen/core/EnableNamedTensor.h`, which defines
the BUILD_NAMEDTENSOR C++ macro. There should be no uses of this left
after this PR, but it is future work to check that

Test Plan:
- run tests

Differential Revision: [D18908442](https://our.internmc.facebook.com/intern/diff/D18908442)

[ghstack-poisoned]

zou3519 commented:
The `pytorch_windows_build` failed here, but a rebuild succeeded: https://app.circleci.com/jobs/github/pytorch/pytorch/3900753 so I am considering that failure to be flaky.
